test/e2e/pivoting_test.go

```diff
@@ -0,0 +1,263 @@
+			return err
+		}
+		for _, machine := range machines.Items {
+			if !strings.EqualFold(machine.Status.Phase, "running") { // Case insensitive comparison
+				return errors.New("Machines cannot be in the Running state")
+			}
+		}
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-machine-running")...).Should(BeNil())
+
+	By("PASSED!")
+}
+
+func configureIronicConfigmap(isIronicDeployed bool) {
+	bmoPath := "BMOPATH"
+	ironicDataDir := "IRONIC_DATA_DIR"
+	ironicConfigmap := fmt.Sprintf("%s/ironic-deployment/keepalived/ironic_bmo_configmap.env", os.Getenv(bmoPath))
+	newIronicConfigmap := fmt.Sprintf("%s/ironic_bmo_configmap.env", os.Getenv(ironicDataDir))
+	backupIronicConfigmap := fmt.Sprintf("%s/ironic-deployment/keepalived/ironic_bmo_configmap.env.orig", os.Getenv(bmoPath))
+	if isIronicDeployed {
+		cmd := exec.Command("cp", ironicConfigmap, backupIronicConfigmap)
+		err := cmd.Run()
+		Expect(err).To(BeNil(), "Cannot run cp command")
+		cmd = exec.Command("cp", newIronicConfigmap, ironicConfigmap)
+		err = cmd.Run()
+		Expect(err).To(BeNil(), "Cannot run cp command")
+	} else {
+		cmd := exec.Command("mv", backupIronicConfigmap, ironicConfigmap)
+		err := cmd.Run()
+		Expect(err).To(BeNil(), "Cannot run mv command")
+	}
+}
+
+func installIronicBMO(targetCluster framework.ClusterProxy, isIronic, isBMO string) {
+	ironicTLSEnable := "IRONIC_TLS_SETUP"
+	ironicBasicAuth := "IRONIC_BASIC_AUTH"
+	ironicHost := "172.22.0.2"
+	bmoPath := "BMOPATH"
+	path := fmt.Sprintf("%s/tools/", os.Getenv(bmoPath))
+	args := []string{
+		isBMO,
+		isIronic,
+		os.Getenv(ironicTLSEnable),
+		os.Getenv(ironicBasicAuth),
+		"true",
+	}
+	env := []string{
+		fmt.Sprintf("IRONIC_HOST=%s", ironicHost),
+		fmt.Sprintf("IRONIC_HOST_IP=%s", ironicHost),
+		fmt.Sprintf("KUBECTL_ARGS=--kubeconfig=%s", targetCluster.GetKubeconfigPath()),
+		"USER=ubuntu",
+	}
+	cmd := exec.Command("./deploy.sh", args...)
+	cmd.Dir = path
+	cmd.Env = append(env, os.Environ()...)
+
+	err := cmd.Start()
+	Expect(err).To(BeNil(), "Fail to deploy Ironic")
+	err = cmd.Wait()
+	Expect(err).To(BeNil(), "Fail to deploy Ironic")
+}
+
+func removeIronicContainers() {
+	ironicContainerList := []string{
+		"ironic-api",
+		"ironic-conductor",
+		"ironic-inspector",
+		"dnsmasq",
+		"mariadb",
+		"ironic-endpoint-keepalived",
+		"ironic-log-watch",
+		"ironic-inspector-log-watch",
+	}
+	// dockerClient, err := docker.NewEnvClient()
+	dockerClient, err := docker.NewClientWithOpts()
+	Expect(err).To(BeNil(), "Unable to get docker client")
+	removeOptions := dockerTypes.ContainerRemoveOptions{
+		Force: true,
+	}
+	for _, container := range ironicContainerList {
+		err = dockerClient.ContainerRemove(ctx, container, removeOptions)
+		Expect(err).To(BeNil(), "Unable to delete the container %s: %v", container, err)
+	}
+}
+func removeIronicDeployment() {
+	deploymentName := os.Getenv("NAMEPREFIX") + "-ironic"
+	ironicNamespace := os.Getenv("IRONIC_NAMESPACE")
```

Comment:
I do also suggest we take all vars like these here and in all other funcs so we could define them at the top once and re-use them (if applicable) later, since some of them redefined twice in different funcs, i.e `IRONIC_NAMESPACE`, `BMOPATH`

Reply:
I agree that some variables are redefined some times in this PR. However, I think it is not too bad since these variables are mostly in type string, not big objects.I kind of reluctant to use global variables if we only need them in one test to avoid unnecessary access to these variables from other tests. Therefore, I would use local variables in this case.